### PR TITLE
Type `result` as `string | object` in `dialog.submitHandler`

### PR DESCRIPTION
**Pull request by MarcinJuraszek**
Addresses https://github.com/OfficeDev/microsoft-teams-library-js/issues/444 in 2.0-preview branch for both the new `dialog.open` and backwards-compatible `tasks.startTask`.

### CODE REVIEW

**Comment by erinha** [comment]
<p>Analyzed commit id : c1df8f709fb1762121015dd5b16c0f4f4a32134d</p><hr><details><summary><b><span style="color: green">⯆</span> @microsoft/bundle-analysis-app:</b> -6 Bytes</summary><table><tr><th>Metric Name</th><th>Baseline Size</th><th>Compare Size</th><th>Size Diff</th></tr><tr><td>main.js</td><td>57.33 KB</td><td>57.32 KB</td><td><span style="color: green">⯆</span> -6 Bytes</td></tr><tr><td>Total Size</td><td>57.33 KB</td><td>57.32 KB</td><td><span style="color: green">⯆</span> -6 Bytes</td></tr></table></details><hr><p>Baseline commit: 256a27a83154f07d1a6087f8f5959ce19d4943f9</p>

**Comment by erinha** [comment]
<p>Analyzed commit id : 2e7e46cb2e423b4590418cd3230b228dc3b1be86</p><hr><details><summary><b><span style="color: green">⯆</span> @microsoft/bundle-analysis-app:</b> -6 Bytes</summary><table><tr><th>Metric Name</th><th>Baseline Size</th><th>Compare Size</th><th>Size Diff</th></tr><tr><td>main.js</td><td>57.34 KB</td><td>57.34 KB</td><td><span style="color: green">⯆</span> -6 Bytes</td></tr><tr><td>Total Size</td><td>57.34 KB</td><td>57.34 KB</td><td><span style="color: green">⯆</span> -6 Bytes</td></tr></table></details><hr><p>Baseline commit: 9835bb8c2b9e89de9b3eb4ef4b11985e945916c3</p>

**Review by TrevorJoelHarris — APPROVED** [review]
:shipit:

**Comment by erinha** [comment]
<p>Analyzed commit id : dbe938e1cf70bac035d0e5fe10b8a46926fbb72e</p><hr><details><summary><b><span style="color: green">⯆</span> @microsoft/bundle-analysis-app:</b> -6 Bytes</summary><table><tr><th>Metric Name</th><th>Baseline Size</th><th>Compare Size</th><th>Size Diff</th></tr><tr><td>main.js</td><td>57.34 KB</td><td>57.34 KB</td><td><span style="color: green">⯆</span> -6 Bytes</td></tr><tr><td>Total Size</td><td>57.34 KB</td><td>57.34 KB</td><td><span style="color: green">⯆</span> -6 Bytes</td></tr></table></details><hr><p>Baseline commit: 8103a864bee9c57df538093125f1707df748e936</p>

**Review by erinha — APPROVED** [review]

File: packages/teams-js/src/public/tasks.ts [hunk 1]
```diff
@@ -19,7 +21,10 @@ export namespace tasks {
    * @param taskInfo - An object containing the parameters of the task module
    * @param submitHandler - Handler to call when the task module is completed
    */
-  export function startTask(taskInfo: TaskInfo, submitHandler?: (err: string, result: string) => void): IAppWindow {
+  export function startTask(
```
[the remark this answers is not in the document]
Comment (MarcinJuraszek) [reply packages/teams-js/src/public/tasks.ts:24]: I would let Yuri and Sachin make the call on that.

**Pull request merged**